spec/integration/transport/ssh_spec.rb

```diff
@@ -469,4 +469,15 @@ def make_target(host_: hostname, port_: port, conf: config)
       expect { ssh.run_command(target, 'whoami') }.to raise_error(/does not have a host/)
     end
   end
+
+  context "user supplied script directory" do
+    let(:config) do
+      mk_config("host-key-check" => false, "sudo-password" => password, "run-as" => "root",
+                user: user, password: password, "script-dir" => "123456", interpreters: { sh: "/bin/sh" })
+    end
+    let(:target) { make_target }
+    it "executes a command on a host", ssh: true do
+      expect(ssh.run_command(target, "pwd")["stdout"]).to eq("/home/bolt\n")
```

Comment:
`run_command` doesn't actually create a tempdir to execute in (https://github.com/puppetlabs/bolt/blob/master/lib/bolt/transport/sudoable.rb#L20 - would have `conn.with_tempdir` if it used a tempdir). You can make a temporary script that prints the working directory and use that to verify that the correct directory is used:

```
  context "user supplied script directory" do
    let(:script_dir) { "123456" }
    let(:config) do
      mk_config("host-key-check" => false, "sudo-password" => password,
                "run-as" => "root", user: user, password: password,
                "script-dir" => script_dir, interpreters: { sh: "/bin/sh" })
    end 
    let(:target) { make_target(conf: config) }

    it "executes a command on a host", ssh: true do
      cmd = 'cd $( dirname $0) && pwd'
      with_tempfile_containing('dir', cmd, '.sh') do |script|
        result = ssh.run_script(target, script.path, nil)
        expect(result.value['stdout']).to eq("/tmp/123456\n")
      end 
    end 
  end 
```

Reply:
You'll also want to pass your config to the `make_target` call, like above.